docs/core/promises.md

```diff
@@ -1,14 +1,80 @@
+that gap.  The Qooxdoo `qx.Promise` is also guaranteed cross platform, so even if you still depend on older browsers such as IE11 which do not have native
+support for Promises built in, you can start using them today. 
+
 Promises are non-trivial to implement properly, so this class is implemented thanks to BluebirdJS from http://bluebirdjs.com/ (also under MIT license)
 
-## Asynchronous Apply Methods
-It is reasonable that any applyXxx method could take some time to complete (including triggering a server roundtrip), in which case it is necessary to delay the firing the changeXxx event until the apply is complete, otherwise the event is fired before the property value is fully applied.
-The return value for apply methods is normally ignored, but with this PR if you return a qx.Promise the setXxx will wait for the promise to be fulfilled before firing the event.
-As setXxx method normally returns the value which has been set, it is not possible to return the promise to the caller - which is a disadvantage because it follows that there could be other processes which depend on the successful, and sequential, completion of the property apply.
+## Introduction
+Promises are really easy use - a lot of the detail and subtleties are beyond the scope of this article, but any documentation you can find on how to use 
+native `Promise` will apply to Qooxdoo promises, all you need to do is substitute `Promise` in other documentation with `qx.Promise`.
+
+Here's a basic example:
+
+```javascript
+function connectToServer() {
+    return new qx.Promise((resolve, reject) => {
+        // Connect to the server asynchronously here; if the server call is successful,
+        //  we call resolve() and handle errors by calling reject().   
+        resolve("Success");
+    });
+}
+
+function main() {
+    connectToServer()
+        .then(result => {
+            console.log("Connecting to server responded: " + result);
+        });
+}
+
+main();
+```
 
-## A new setXxxAsync Method
-For properties which are known in advance to have asynchronous apply methods, you can enable a new ```setXxxAsync``` method by setting the ```async: true``` value on the property configuration; the normal property accessors operate in the same way that they always did and remain backward compatible, it’s just that there is ```setXxxAsync``` method that returns a ```qx.Promise```. Note that it always returns a promise, even if the apply method is not asynchronous and does not return anything - this is so that setXxxAsync always operates consistently regardless of how it is implemented.
+In that example, the console will print:
+```
+Connecting to server responded: Success
+```
+
+Promises are the code that makes `async` and `await` work in ES6 (which you can also use cross platform with Qooxdoo, provided you use the qooxdoo-compiler):
+
+```javascript
+async function connectToServer() {
+    await new qx.Promise((resolve, reject) => {

```

Comment:
An async function returns a promise. I think this should be `return new qx.Promise(...` rather than `await new qx.Promise(...`